packages/phasor/src/surface.ts

```diff
@@ -141,10 +146,19 @@ export class SurfaceManager {
   }
 
   pickTop(x: number, y: number): PhasorElement | null {
-    const results = this.pick(x, y);
+    const results = this._pickByPoint(x, y);
     return results[results.length - 1] ?? null;
   }
 
+  pickByBound(bound: IBound): PhasorElement[] {
+    const candidates = this._renderer.gridManager.search(bound);
+    const picked = candidates.filter((element: PhasorElement) => {
+      return contains(bound, element) || intersects(bound, element);
+    });
```

Comment:
I'm not sure to test bound here. Because different shapes may have different intersect logic.